### ISSUE TITLE
AssertionError when using ProactorEventLoop

### ISSUE THREAD

**Issue by AGSPhoenix** [Rapptz/discord.py#859]
The ProactorEventLoop is the only event loop on Windows that supports subprocesses, but trying to use it with Discord.py results in it spitting out an error and soft locking. 

Error message:
```
Exception in callback _ProactorReadPipeTransport._loop_reading(<_OverlappedF...op_reading()]>)
handle: <Handle _ProactorReadPipeTransport._loop_reading(<_OverlappedF...op_reading()]>)>
Traceback (most recent call last):
  File "c:\python36\Lib\asyncio\events.py", line 127, in _run
    self._callback(*self._args)
  File "c:\python36\Lib\asyncio\proactor_events.py", line 190, in _loop_reading
    data = fut.result()  # deliver data later in "finally" clause
asyncio.base_futures.InvalidStateError: Result is not ready.
Exception in callback _ProactorReadPipeTransport._loop_reading(<_OverlappedF...\x92\xfd\xe9'>)
handle: <Handle _ProactorReadPipeTransport._loop_reading(<_OverlappedF...\x92\xfd\xe9'>)>
Traceback (most recent call last):
  File "c:\python36\Lib\asyncio\events.py", line 127, in _run
    self._callback(*self._args)
  File "c:\python36\Lib\asyncio\proactor_events.py", line 188, in _loop_reading
    self._closing)
AssertionError
```

Source code to repro is just the README.md example modified to use Proactor before the Discord Client is initialized.
```
import discord

import asyncio, sys

if sys.platform == 'win32':
    loop = asyncio.ProactorEventLoop()
    asyncio.set_event_loop(loop)

client = discord.Client()

@client.event
async def on_ready():
    print('Logged in as')
    print(client.user.name)
    print(client.user.id)
    print('------')

@client.event
async def on_message(message):
    if message.content.startswith('!test'):
        counter = 0
        tmp = await client.send_message(message.channel, 'Calculating messages...')
        async for log in client.logs_from(message.channel, limit=100):
            if log.author == message.author:
                counter += 1

        await client.edit_message(tmp, 'You have {} messages.'.format(counter))
    elif message.content.startswith('!sleep'):
        await asyncio.sleep(5)
        await client.send_message(message.channel, 'Done sleeping')

client.run('token')
```
I'm running Windows 7 with Python 3.6.

**Comment by khazhyk** [comment]
How frequent are we talking here? Every day? Every minute?

**Comment by AGSPhoenix** [comment]
Almost immediately after running it; about 5 seconds on my development machine.

**Comment by khazhyk** [comment]
I can't repro this on w10 py3.5. My bot's been running pretty happily for about 10 minutes now.

**Comment by khazhyk** [comment]
I can repro this with py 3.6 on w10. The error however looks to be in a pretty core area, this looks like a bug in ProactorEventLoop to me

**Comment by Harmon758** [comment]
For posterity, this was a [Python bug](https://bugs.python.org/issue26819) that was fixed in Python 3.7 and 3.6.6.
